src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,81 @@
+package duke.task;
+import duke.task.Task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * The deadline class is a task class that has an additional parameter by.
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
+
+    /**
+     * Basic constructor.
+     * @param description The description of the task
+     * @param by The time the task should be completed by
+     */
+    public Deadline(String description, LocalDateTime by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Overloaded constructor where the isDone field can be set.
+     * @param description The description of the task.
+     * @param by The time the task should be completed by.
+     * @param isDone Whether the task has been completed.
+     */
+    public Deadline(String description, LocalDateTime by, boolean isDone) {
+        super(description,
+                isDone);
+        this.by = by;
+    }
+
+    /**
+     * Overridden toString method to print the object.
+     * @return String representation of the object.
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + this.formatDate() + ")";
+    }
+
+
+    /**
+     * Date formatter that formats it to dd/MM/yyyy.
+     * @return String of the formatted date.
+     */
+    public String formatDate() {
+
+        if (by.getHour() == 0 && by.getMinute() == 0) {
+            return by.format(DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+        } else {
+            return this.saveDate();
+        }
```

Comment:
Maybe could consider omitting the `else` keyword? Might look a bit cleaner if you prefer